dnSpy/dnSpy.Contracts.DnSpy/MVVM/BaseVMComparer.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace dnSpy.Contracts.MVVM {
+	/// <summary>
+	/// Base class for view-model property comparers
+	/// </summary>
+	public abstract class BaseVMComparer<T> : IComparer<T> where T : class {
+		/// <summary>
+		/// Ordering info used for sorting collections using this comparer
+		/// </summary>
+		public SortInfo SortInfo { get; set; }
+
+		/// <summary>
+		/// Compares two view-model objects by a given property
+		/// </summary>
+		/// <param name="x">First view-model (not null)</param>
+		/// <param name="y">Second view-model (not null)</param>
+		/// <param name="propertyName">View-model property name</param>
+		protected abstract int CompareByPropertyImpl(T x, T y, string propertyName);
+
+		/// <summary>
+		/// Compares two given view-model object using provided SortInfo
+		/// </summary>
+		public virtual int Compare(T x, T y) {
```

Comment:
This method should be sealed. There's a bug in the overridden method. It checks for 0 and if it's 0 it uses the Order column. But it never uses the `Direction` property. Perhaps add an abstract CompareDefault() method that returns the default sort order.